File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/8.go
package main
import "fmt"

func divisivelPor3Zeros(s string) bool {
    countZeros := 0
    for _, char := range s {
        if char == '0' {
            countZeros++
        }
    }
    return countZeros % 3 == 0
}

func main() {
    fmt.Println(divisivelPor3Zeros("000111"))  // True
    fmt.Println(divisivelPor3Zeros("0011"))    // False
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/8.java
public class AFNDivisivelPor3Zeros {
    public static boolean divisivelPor3Zeros(String s) {
        int countZeros = 0;
        for (char c : s.toCharArray()) {
            if (c == '0') countZeros++;
        }
        return countZeros % 3 == 0;
    }

    public static void main(String[] args) {
        System.out.println(divisivelPor3Zeros("000111"));  // True
        System.out.println(divisivelPor3Zeros("0011"));    // False
    }
}
